### PR TITLE
Add generic search engine menu item

### LINKED ISSUE #155: Add other search engines
As security-minded person, it would be good to allow searching by other search engines eg DuckDuckGo.
If using DDG, the url is of the form: `https://duckduckgo.com/?q=${searchTerm}`

Other users I have seen also use Bing in some cases.

### PR DESCRIPTION
**Pull request by hns258**
Adding an `addSearchWithOther` option that displays a `Search with {searchEngine}` menu item when an object containing the `title` and `url` of a search engine is provided.

Just wanted to mention a few items I wasn't sure about (still relatively new to js):
- Considered creating an interface or typedef for the object `addSearchWithOther` expects but wasn't sure if that'd be "overkill" since a developer has to jump to another section to see the properties needed. I can update to one of those if preferred, though.
- Was on the fence for the option name. I'm willing to change to `showSearchWithOther`, `showOtherSearchEngine`, or anything else. I just wasn't sure if there might be confusion with the others starting with `show...` being booleans (but maybe it's clear enough with the readme and types file?) so for now decided to err on "more clarity" despite looking a bit out of place. 
- Considered doing some validation on the `url` provided but due to the following, it seemed safe to assume that for ~95% of the use cases, the `url` would be "clean" so decided not to but can update, if preferred.
  - the URLs should be from a small set of sites
  - this module is used by developers 
  - the check seems to need a helper function or 3rd party lib (didn't want to unnecessarily clutter the code)

Fixes #155

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -70,6 +73,16 @@ const create = (win, options) => {
 					electron.shell.openExternal(url.toString());
 				}
 			}),
+			searchWithOther: decorateMenuItem({
+				id: 'searchWithOther',
+				label: `Search &with ${addSearchWithOtherHasValues ? options.addSearchWithOther.title : ''}`,
+				visible: hasText,
+				click() {
+					const url = new URL(addSearchWithOtherHasValues ? options.addSearchWithOther.url : '');
+					url.searchParams.set('q', props.selectionText);
```
Comment (sindresorhus) [inline index.js:82]: You are incorrectly assuming all search engines use `q`. Please follow what was discussed in the original issue.

**Comment by sindresorhus** [comment]
Make sure you have updated all the required places. See earlier submission for what to update:
- https://github.com/sindresorhus/electron-context-menu/commit/c50c1f5bc81fd07926a0cd37f9dcaba6fb014199
- https://github.com/sindresorhus/electron-context-menu/commit/3efba17dbe44c94f28dc38d8548a97cb80c260ca

**Comment by sindresorhus** [comment]
> Considered creating an interface or typedef for the object addSearchWithOther expects but wasn't sure if that'd be "overkill" since a developer has to jump to another section to see the properties needed. I can update to one of those if preferred, though.

I don't think that's needed.

> Considered doing some validation on the url provided but due to the following, it seemed safe to assume that for ~95% of the use cases, the url would be "clean" so decided not to but can update, if preferred.

I don't think validation is needed.

File: index.d.ts [hunk 2]
```diff
@@ -185,6 +192,34 @@ declare namespace contextMenu {
 		*/
 		readonly showSearchWithGoogle?: boolean;
 
+		/**
+		Add a `Search with {searchEngine}` menu item when right-clicking text.
+
+		This allows the use of a search engine besides Google (e.g., Bing and DuckDuckGo). The `title` and `url` of the desired search engine need to be provided. The `{searchEngine}` placeholder will be replaced by `title`.
+		@property {string} title Title/name of search engine
+		@property {string} url URL of search engine (with query syntax used in Chrome)
```
Comment (sindresorhus) [inline index.d.ts:200]:
```suggestion
		@property {string} url - The URL of the search engine (with query syntax used in Chrome).
```

**Review by sindresorhus — CHANGES_REQUESTED** [review]

File: index.js (hunk 1 above)
Comment (hns258) [reply index.js]: Sorry, I misread the doc, updating now

**Comment by hns258** [comment]
Realized this PR probably isn't the preferred solution to the referenced issue, closing it. Apologies for any wasted time.

**Pull request closed without merging**